rpcserver.go

```diff
@@ -5563,17 +5563,15 @@ func handleTxFeeInfo(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (
 // handleValidateAddress implements the validateaddress command.
 func handleValidateAddress(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
 	c := cmd.(*dcrjson.ValidateAddressCmd)
-
 	result := dcrjson.ValidateAddressChainResult{}
 	addr, err := dcrutil.DecodeAddress(c.Address)
 	if err != nil {
 		// Return the default value (false) for IsValid.
 		return result, nil
 	}
 
-	result.Address = addr.EncodeAddress()
-	result.IsValid = true
-
+	result.Address = c.Address
```

Comment:
This shouldn't be set when it's for the wrong network.

Notice the difference between `dcrwallet` and `dcrd` below.

dcrd:

```
$ dcrctl validateaddress TsYEVnkLPSniXpqUNqPZKEktCLkBzjXjK6n
{
  "isvalid": false,
  "address": "TsYEVnkLPSniXpqUNqPZKEktCLkBzjXjK6n"
}
```

dcrwallet:
```
$ dcrctl --wallet validateaddress TsYEVnkLPSniXpqUNqPZKEktCLkBzjXjK6n
{
  "isvalid": false,
}
```

Reply:
The following would result in the desired behavior:

```Go
	if err != nil || !addr.IsForNet(s.server.chainParams) {
		// Return the default value (false) for IsValid.
		return result, nil
	}

	result.Address = c.Address
	result.IsValid = true
	return result, nil
```